src/main/java/com/cloudbees/hudson/plugins/folder/computed/EventOutputStreams.java

```diff
@@ -0,0 +1,289 @@
+
+            private void lazyFlush() throws IOException {
+                long end = System.nanoTime();
+                if (index >= buf.length || end - start > flushIntervalNanos) {
+                    start = end;
+                    if (index == 0) {
+                        return;
+                    }
+                    int count = index;
+                    for (int i = index - 1; i >= 0; i--) {
+                        if (buf[i] == '\n') {
+                            count = i + 1;
+                            break;
+                        }
+                    }
+                    if (index == count) { // send the whole buffer
+                        offer(buf.clone());
+                        index = 0;
+                    } else { // send a partial buffer up to the last newline
+                        offer(Arrays.copyOf(buf, count));
+                        System.arraycopy(buf, count, buf, 0, index - count);
+                        index -= count;
+                    }
+                }
+            }
+
+            @Override
+            public void write(byte[] data, int off, int len) throws IOException {
+                if (data == null) {
+                    throw new NullPointerException();
+                }
+                if (index + len <= buf.length) {
+                    System.arraycopy(data, off, buf, index, len);
+                    index += len;
+                    lazyFlush();
+                } else {
+                    while (len > 0) {
+                        int l = Math.min(len, buf.length - index);
+                        write(data, off, l);
+                        off += l;
+                        len -= l;
+                    }
+                }
+            }
+
+            @Override
+            public void flush() throws IOException {
+                start = System.nanoTime();
+                if (index > 0) {
+                    offer(Arrays.copyOf(buf, index));
+                    index = 0;
+                }
+            }
+
+            @Override
+            public void close() throws IOException {
+                flush();
+                offer(null); // force a flush to disk
+            }
+        };
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void close() throws IOException {
+        offer(null);
+    }
+
+    /**
+     * Supplies the current output file destination. We use indirection so that when used from a
+     * {@link FolderComputation} the containing {@link Folder} can be moved without keeping the events log open.
+     */
+    public static abstract class OutputFile {
+        /**
+         * Returns the file that output is being sent to.
+         *
+         * @return the file that output is being sent to.
+         */
+        @NonNull
+        public abstract File get();
+
+        /**
+         * Returns {@code true} if the output file can be written to now, {@code false} if the write should be delayed.
```

Comment:
Yes, but I cannot see another way... plus the move impl we know off will retry if there is a locked file so whoop the buffer saves us ;-)